api/edge_api/identities/views.py

```diff
@@ -97,44 +95,43 @@ def initial(self, request, *args, **kwargs):
 
         super().initial(request, *args, **kwargs)
 
-    def _get_search_function_and_value(
-        self,
-        search_query: str,
-    ) -> typing.Tuple[typing.Callable, str]:
-        if search_query.startswith('"') and search_query.endswith('"'):
-            return self.dynamo_identifier_search_functions[
-                "EQUAL"
-            ], search_query.replace('"', "")
-        return self.dynamo_identifier_search_functions["BEGINS_WITH"], search_query
+    def get_serializer_class(self):
+        if self.action in ("update", "partial_update"):
+            return EdgeIdentityUpdateSerializer
+        return EdgeIdentitySerializer
 
     def get_object(self):
+        # TODO: should this return an EdgeIdentity object instead of a dict?
```

Comment:
I'm leaving this here as a todo because I didn't want to bloat this PR with this work, but I do think this should be refactored.

Reply:
Maybe make a GitHub issue for it? I find code TODOs often get forgotten about over time.

Reply:
#4578